### PR TITLE
Adding Azure EventHubs as output

### PR DESCRIPTION
**Pull request by arminc**
This adds Azure EventHubs as an output option. The current implementation assumes usage of Azure "aad-pod-identity".

Extra: It adds priorityClassName to the helm chart.

### CODE REVIEW

**Comment by Issif** [comment]
Thanks for that great MR. I'm doing a review ASAP, even if I don't know how Azure works 😉

**Comment by Issif** [comment]
@arminc We've integrated an automatic system, poiana for handling merge requests and be consistent with the whole falcosecurity organization. It implies you sign-off all your commits, can you do that pleaseN

**Comment by arminc** [comment]
Done

File: deploy/helm/falcosidekick/templates/deployment.yaml [hunk 1]
```diff
@@ -18,8 +18,12 @@ spec:
       labels:
         app.kubernetes.io/name: {{ include "falcosidekick.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
+        aadpodidbinding: {{ include "falcosidekick.fullname" . }}
```
Comment (Issif) [inline deploy/helm/falcosidekick/templates/deployment.yaml:21]: Can you add a condition like `{{- if and .Values.config.azure.podIdentityClientID .Values.config.azure.podIdentityName -}}`

File: outputs/azure.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		Config:          config,
+		Stats:           stats,
+		StatsdClient:    statsdClient,
+		DogstatsdClient: dogstatsdClient,
+	}, nil
+}
+
+// EventHubPost posts event to Azure Event Hub
+func (c *Client) EventHubPost(falcopayload types.FalcoPayload) {
+	log.Printf("[INFO] : Try sending event")
```
Comment (Issif) [inline outputs/azure.go:27]:
```suggestion
	log.Printf("[INFO]  : Try sending event")
```

File: outputs/azure.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	hub, err := eventhub.NewHubWithNamespaceNameAndEnvironment(c.Config.Azure.EventHub.Namespace, c.Config.Azure.EventHub.Name)
+	if err != nil {
+		c.Stats.AzureEventHub.Add("error", 1)
+		log.Printf("[ERROR] : %v EventHub - %v\n", c.OutputType, err.Error())
+		return
+	}
+	log.Printf("[INFO] : %v EventHub - Hub client created\n", c.OutputType)
```
Comment (Issif) [inline outputs/azure.go:34]:
```suggestion
	log.Printf("[INFO]  : %v EventHub - Hub client created\n", c.OutputType)
```

File: outputs/azure.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		return
+	}
+	err = hub.Send(ctx, eventhub.NewEvent(data))
+	if err != nil {
+		c.Stats.AzureEventHub.Add("error", 1)
+		log.Printf("[ERROR] : %v EventHub - %v\n", c.OutputType, err.Error())
+		return
+	} else {
+		c.Stats.AzureEventHub.Add("ok", 1)
+		log.Printf("[INFO] : Succesfully sent event")
```
Comment (Issif) [inline outputs/azure.go:51]:
```suggestion
		log.Printf("[INFO]  : Succesfully sent event")
```

File: stats.go [hunk 5]
```diff
@@ -38,6 +38,7 @@ func getInitStats() *types.Statistics {
 		Statsd:        expvar.NewMap("outputs.statsd"),
 		Dogstatsd:     expvar.NewMap("outputs.dogstatsd"),
 		Webhook:       expvar.NewMap("outputs.webhook"),
+		AzureEventHub: expvar.NewMap("outputs.eventhub"),
```
Comment (Issif) [inline stats.go:41]:
```suggestion
		AzureEventHub: expvar.NewMap("outputs.azureeventhub"),
```
It's ugly, I agree, but clear.

**Comment by Issif** [comment]
@arminc That's great. Can you add more details in README for telling this output only works if `falcosidekick` is running in an AKS Pod with add label. Thanks.

**Comment by Issif** [comment]
@arminc As my last comments are simple, I approved because feature seems working for me.

**Pull request merged**